CommandLineTool.yml

```diff
@@ -238,6 +239,31 @@ $graph:
         return a string or an array of strings, which will then be evaluated as
         one or more glob patterns.  Must only match and return files which
         actually exist.
+
+        If the value of glob is a relative path pattern (does not
+        begin with a slash '/') then it is resolved relative to the
+        output directory.  If the value of the glob is an absolute
+        path pattern (it does begin with a slash '/') then it must
+        refer to a path within the output directory.  It is an error
+        if any glob, either relative path containing '..' or an
+        absolute path, resolves to a path outside the output
+        directory.
+
+        A glob may match a file within the output directory which is
+        actually a symlink to another file.  It is an error if a
+        symlink in the output directory refers to any file or
+        directory outside the output directory, except for the case of
+        files staged by InitialWorkDirRequirement and only if staging
+        was implementated using symlinks to the original input files.
+
+        A glob which refers to a symlink which is relative within the
+        output directory should produce a File object which has the
+        `location` of the glob path (the symlink is not dereferenced).
```

Comment:
Can you share the motivation for this?

Reply:
The principal of least surprise.  Also, if symlinks are not supported by the storage system, the symlink might be converted to a regular file.

Reply:
Symlinks are part of POSIX ...

Reply:
Yes, but several implementations upload the output directory to cloud storage and do the globbing separately.

Reply:
I can't think of any circumstance that a researcher want to get a symlink "file" and not the actual contents. The choice to use an object store is on the implementer; they can traverse the symlink for the user, if needed.

Reply:
in summary: remove 'location' and `symlink is not dereferenced'; the key is that the basename matches the name of the symlink

Reply:
Looking at 

```
If the value of the glob is an absolute path pattern (it does begin with a slash '/') then it must refer to a path within the output directory.
```

Wouldn't it be simpler to just not accept absolute paths. Looks like we are - correctly - restricting access to the working (output) directory. This does not let people collect outputs from arbitrary directories in the container, which is ok - but there is a case for it. Right now for some tools I tend to have a post processing step involve moving files from where they are produced to the output directory tree.

In summary :) I suggest that we either let people glob from arbitrary locations, or only have glob expressions that are relative to the output directory.

Reply:
@tetron was this change put in (do you agree with it). I agree with @mr-c on this.

Reply:
From video chat discussion:  `location` can be whatever the platform wants, the important thing is that `basename` is consistent with the *symlink* name, and not the name of the file the link points to.